pkg/terraform/util.go

```diff
@@ -34,6 +34,13 @@ func Configure(assetDir, renderedBackend string) error {
 		return errors.Wrapf(err, "Failed to create required terraform directory")
 	}
 
+	// TODO: Don't deploy self-hosted kubelet when `disable_self_hosted_kubelet`
+	// knob is off.
+	// TODO: Deploy calico-host-protection only on Packet.
+	if err := install.PrepareChartDirAt(filepath.Join(assetDir, "cluster-assets", "charts")); err != nil {
```

Comment:
Perhaps unpacking platform-specific manifests can be done in `Initialize()` method of `Platform` interface? This code already deals with creating `cluster.tf` Terraform file.

To be honest `terraform` package seems to be a poor choice for unpacking the assets. IMO `terraform` package should only provide Terraform-related functionality and `Configure` already does too much.

I'd do the unpacking in `cli/cmd/cluster.go#initialize()`, as we don't have a better place to do that right now.

Reply:
Yes, it makes absolutely no sense to me to extract assets in `pkg/terraform`, what more that the extraction happens in `util.go` :disappointed:
I was trying to keep the scope of this PR small, but oh well, I guess we can rename it and possibly fix the diff bug in a separate PR.

Reply:
I've changed the assets handling along the lines you've proposed. I've tried a symlink-based approach where each platform has its own charts directory with symlinks to a `common` directory as well as platform-specific charts. I've also put the self-hosted kubelet in an `optional` directory from which we read charts based on knobs in the cluster config.